processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1098,6 +1098,11 @@ public static File makeTimeFile(File dir, ByteOrder order)
 
   public static File makeMetricFile(File dir, String metricName, ByteOrder order)
   {
-    return new File(dir, String.format("met_%s_%s.drd", metricName, order));
+    return new File(dir, sanitizeFileName(String.format("met_%s_%s.drd", metricName, order)));
+  }
+
+  public static String sanitizeFileName(String fileName)
+  {
+    return fileName.replace(File.separator, "↗");
```

Comment:
null character and / seem to be the only ones posing problem on most common filesystems. Substitution will not solve all edge cases, since you can always create colliding dimensions. Technically we also need to worry about the file length limit if we want to support Windows. If we want to create 'POSIX portable filenames' then we should restrict to `[A–Z] [a–z] [0–9] . _ -`.

I would probably just pick first 248 characters of the dimension name, replace null, and / with _ and append `-` plus the first 6 characters of the sha1 digest.